overlord/snapstate/snapmgr.go

```diff
@@ -289,7 +289,18 @@ func (m *SnapManager) doClearSnapData(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
-	return m.backend.RemoveSnapData(info)
+	if err = m.backend.RemoveSnapData(info); err != nil {
+		return err
+	}
+
+	// Only remove data shared between versions if this is the last version
```

Comment:
Okay, the answer is that we'll need a separate task that kills the data for all snaps at once. If we intentionally want to reset the data for a snap, it should do so for all snaps rather than just the current revision. The experience would probably be awkward otherwise.

So this is fine for now, thanks.